### PR TITLE
Add dynamic metrics

### PR DESCRIPTION
**Pull request by jburns24**
This PR introduces quite a bit of change all with the goal of creating a central location for bootcamp metadata and laying the foundation for generating visualizations off the metadata so we can answer business questions. I will highlight the major.

1) Introduces a pre-commit hook for centralizing bootcamp metadata

I made a choice that metadata should live with the bootcamp section. This feels like the location that will be least likely to be forgotten and thus has the best chance of staying accurate. In order to simplify the client side logic of aggregating the metadata and avoid maintaining things like a list of subsections where meta data lives. I opted for a pre-commit hook that looks at any staged `.md` files and checks them for metadata. This metadata is then aggregated into a master record on the root .md file `README.md`. This master record is to be maintained by the pre-commit hook. For this pr there are many features missing and the hook really only handles adding or changing metadata. It does not handle things like deleting sections or renaming sections. I would like to address these cases in a follow up pr.

2) Introduces Webpack as a module bundler

Prior to this pr we relied on importing browser bundles in index.js. Early on in development I kept finding myself wanting to use npm packages on the client side. Specifically for packages that did not publish a browser bundle. We could have minified and published a bundle on CDN or we could have introduced a bundler like webpack. I picked the latter as we can now leverage any npm package on the front end. We also get the benifit of babble on our production js which will increase the browser support and minification which will help with asset size. This does introduce a new build dependency though

3) Modifies the deploy process

The bootcamp used to rely on some blackbox deploy magic from github. Now it uses a GitHub Action primarily so we can support the new build dependency. But also this will give us more control over how the bootcamp is deployed. 

4) Adds a dynamic graph 

This is what it was all about. With metadata in place and a bundler to leverage npm packages we can now generate graphs and other visualizations off our metadata.

Here is what the new graph looks like on the home page

image

## *Warning*

The GHA changes have not been tested!

### CODE REVIEW

File: package-lock.json
Comment (dreweth) [inline package-lock.json:1]: Babel adds 4k+ lines of dependencies? Wow

File: .github/workflows/static.yml [hunk 1]
```diff
@@ -27,17 +27,33 @@ jobs:
...
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Run webpack build
+        run: npx webpack --config webpack.prod.js
+
```
Comment (jburns24) [inline .github/workflows/static.yml:47]: This has not been tested. At least according to the readme for `v3` of the setup-node they mention version 18 as a major version they support so we should be good there. I tested on node version 19 but dont think I am doing anything that broke between those major versions.

File: package-lock.json
Comment (jburns24) [reply package-lock.json:1]: Well there were a few dependencies added. And it appears they have many transative dependencies. Some more details on the new stuff.


Dev Dependencies
--------------------
Webpack (and webpack helper packages) -> bundler
Babel (and helper packages) -> js transcompiler
simple-git -> for pre-commit hook making interacting with git much easier!

Prod Dependencies
--------------------
chart.js -> it's in the name
charjs-chart-wordcloud -> not being used in this PR but I want a wordcloud of the technologies we cover
gray-matter -> most popular package for parsing front matter
node-stdlib-browser -> makes some stdlib stuff in node accessible client side. gray-matter depended on `Buffer` which is not a thing client side ootb

Comment (dreweth) [reply package-lock.json:1]: Appreciate the detailed and yet concise summary.

**Review by dreweth — APPROVED** [review]
What can I saw about a 1,000+ line PR other than, LGTM 🥇

**Pull request merged**